Keas.Mvc/ClientApp/components/Spaces/SpacesList.tsx

```diff
@@ -11,38 +12,92 @@ interface IProps {
 
 export default class SpacesList extends React.Component<IProps, {}> {
     public render() {
-        let space = null;
-        if(!!this.props.spaces && this.props.spaces.length > 0)
-        {
-            space = this.props.spaces.map(x => (
-                <SpacesListItem
-                    key={x.space.roomKey}
-                    spaceInfo={x}
-                    showDetails={this.props.showDetails}
-                />
-            ));
-        }
-        else
-        {
-            space = (<tr><td colSpan={8}>No Spaces Were Found</td></tr>)
-        }
-
         return (
-            <table className="table">
-                <thead>
-                    <tr>
-                        <th>Building</th>
-                        <th>Floor</th>
-                        <th>Room</th>
-                        <th>Name</th>
-                        <th>Keys</th>
-                        <th>Equipment</th>
-                        <th>Workstations</th>
-                        <th>Actions</th>
-                    </tr>
-                </thead>
-                <tbody>{space}</tbody>
-            </table>
+            <ReactTable
+                data={this.props.spaces}
+                filterable={true}
+                minRows={1}
+                columns = {[
+                    {
+                        Header: "Room",
+                        accessor: (row) => row.space.roomNumber + " " + row.space.bldgName,
+                        id: "room",
+                        filterMethod: (filter, row) => {
+                            return row[filter.id].toLowerCase().indexOf(filter.value.toLowerCase()) !== -1;
```

Comment:
Double checking here but do you need to check existence of `row[filter.id]`?  Seems like everywhere else in the PR you are doing a check before accessing .toLowerCase() but not in this one

Reply:
@srkirkland this should always exist, whereas "Room Name" isn't required--but I added it just to be safe